apps/site/assets/ts/schedule/components/line-diagram/LineDiagramWithStops.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { ReactElement, useEffect } from "react";
+import { hasBranchLines } from "./line-diagram-helpers";
+import Diagram from "./graphics/Diagram";
+import StopListWithBranches from "./StopListWithBranches";
+import { CommonLineDiagramProps } from "./__line-diagram";
+import useStopPositions, { RefList } from "./graphics/useStopPositions";
+import StopCard from "./StopCard";
+
+export const StopRefContext = React.createContext<[RefList, () => void]>([
+  {},
+  () => {}
+]);
+
+const LineDiagramWithStops = (
+  props: CommonLineDiagramProps
+): ReactElement<HTMLElement> => {
+  const { stops, handleStopClick, liveData } = props;
+
+  // create a ref for each stop - we will use this to track the location of the stop so we can place the line diagram bubbles
+  const [stopRefsMap, updateAllStopCoords] = useStopPositions(stops);
+
+  useEffect(
+    () => {
+      window.addEventListener("resize", updateAllStopCoords);
+      updateAllStopCoords();
+      return () => window.removeEventListener("resize", updateAllStopCoords);
+    },
+    [updateAllStopCoords]
+  );
+
+  return (
+    <StopRefContext.Provider value={[stopRefsMap, updateAllStopCoords]}>
```

Comment:
The first value is used in each of the `StopCard` components (as it contains the refs), and the second is used in `ExpandableBranch` (so that we can manually trigger updating all the offsetHeight and offsetTop values for the stop bubbles when opening/closing a branch);

Reply:
I see now, thanks for helping me understand!